grafana-plugin/src/pages/users/Users.tsx

```diff
@@ -74,18 +76,20 @@ class Users extends React.Component<UsersProps, UsersState> {
     const { usersFilters, page } = this.state;
     const { userStore } = store;
 
-    if (!isUserActionAllowed(UserActions.UserSettingsWrite)) {
+    if (!isUserActionAllowed(this.REQUIRED_PERMISSION_TO_VIEW_USERS)) {
+      this.setState({ initialUsersLoaded: true });
       return;
     }
 
     LocationHelper.update({ p: page }, 'partial');
-    return await userStore.updateItems(usersFilters, page);
+    await userStore.updateItems(usersFilters, page);
+
+    this.setState({ initialUsersLoaded: true });
   };
 
   componentDidUpdate(prevProps: UsersProps) {
-    if (!this.initialUsersLoaded && isUserActionAllowed(UserActions.UserSettingsWrite)) {
+    if (!this.state.initialUsersLoaded) {
       this.updateUsers();
-      this.initialUsersLoaded = true;
```

Comment:
this needs to be stored as state, not a class attribute. I believe the problem was that:
```tsx
this.initialUsersLoaded = true;
```
would not trigger a re-render.